apps/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -76,7 +123,40 @@ export class PackageChangeAnalyzer {
     return projectState;
   }
 
-  private async _getData(terminal: Terminal): Promise<Map<string, Map<string, string>> | undefined> {
+  /**
+   * Gets a list of projects that have changed in the current state of the repo
+   * when compared to the specified branch.
+   */
+  public async *getChangedProjectsAsync(
+    targetBranch: string,
+    shouldFetch: boolean = false
+  ): AsyncIterable<RushConfigurationProject> {
+    const changedFolders: string[] | undefined = this._git.getChangedFolders(targetBranch, shouldFetch);
```

Comment:
When `shouldFetch=true`, I noticed that the underlying code path calls `console.log()`:

https://github.com/microsoft/rushstack/blob/9d8d25e6b46737efc5a46081cd809d95e1fff275/apps/rush-lib/src/logic/Git.ts#L461-L467

Should the options parameter include a `terminal`?